default/python/AI/ProductionAI.py

```diff
@@ -1091,6 +1091,42 @@ def generate_production_orders():
         else:
             warning("Failed enqueueing %s at planet %s, got result %d", building_name, planet, res)
 
+    if tech_is_complete(GRO_GAIA_TRANS):
+        terraformers = ["BLD_GAIA_TRANS", "BLD_TERRAFORM"]
+    elif tech_is_complete(GRO_TERRAFORM):
+        terraformers = ["BLD_TERRAFORM"]
+    else:
+        terraformers = []
+    if terraformers:
+        queued_locs = [element.locationID for element in production_queue if (element.name in terraformers)]
+        candidates = []
+        for pid in state.get_inhabited_planets():
+            planet = universe.getPlanet(pid)
+            if (planet.type not in [fo.planetType.gasGiant, fo.planetType.asteroids]
+                and planet.currentMeterValue(fo.meterType.population) == planet.currentMeterValue(fo.meterType.targetPopulation)
+                and pid not in queued_locs
+                and aistate.systemStatus.get(planet.systemID, {}).get('fleetThreat', 1) == 0
+                and "GAIA_SPECIAL" not in planet.specials
+                and "NO_TERRAFORM" not in fo.getSpecies(planet.speciesName).tags
+                and not [bldg for bldg in planet.buildingIDs if universe.getBuilding(bldg).buildingTypeName in terraformers]):
+                    candidates.append([pid, planet.currentMeterValue(fo.meterType.population)])
+        if candidates:
+            candidates.sort(key=itemgetter(1), reverse=True)
+            for bldg in terraformers:
+                for pid in candidates:
```

Comment:
it is not pid, it is a pair of pid and population.  

You can use unpacking:  `for pid, _ in candidates: planet = universe.getPlanet(pid)`